calicoctl.py

```diff
@@ -307,19 +289,13 @@ def node(ip, node_image, ip6=""):
     node_docker_client = docker_client
 
     enable_socket = "NO"
-
-    try:
-        if DOCKER_OPTIONS in open(DOCKER_DEFAULT_FILENAME).read():
-            enable_socket = "YES"
-            # At this point, docker is listening on a new port but powerstrip
-            # might not be running, so docker clients need to talk directly to
-            # docker.
-            node_docker_client = docker.Client(version=DOCKER_VERSION,
-                                               base_url="unix://%s" % REAL_SOCK)
-    except IOError:
-        # Can't find Docker config file.  Don't attempt to change anything.
-        # Just carry on.
-        pass
+    if docker_restarter.is_using_alternative_socket():
```

Comment:
This code would be more clear if you moves the two above lines into an `else:` branch of this `if:`